### DISCUSSION TITLE
Log scale in R help :)

### DISCUSSION THREAD

**Discussion by drlawson** [RobertsLab/resources#1312] [Q&A] [answered]
Hi all, I made this figure and table to represent metadata from the nightingales datasheet. I want the y axis to be logarithmic so you can actually see the values for each primary taxa. However, when i add the log scale on the y axis, the data seems to be incorrect (shows P_generosa as having more than O_lurida which is not correct). Anyone know how to solve this? Or am I interpreting this incorrectly? Thanks :)

My code for the table is 
>data(nightingales)
>nighttable <- (nightingales %>% group_by(Primary_taxa, Molecule) %>% summarize(n = n(), .groups = "keep"))
>options(knitr.kable.NA = '0')
>nighttable %>% pivot_wider(names_from = Molecule, values_from = n) %>% kbl(digits=2) %>% kable_classic(full_width = F, html_font = "Cambria") 

My code for the first figure is 
>cols <- c("DNA" = "darkblue", "RNA" = "cornflowerblue", "Unknown" = "darkgray")
>ggplot(data = nighttable, aes(x = Primary_taxa, y = n, fill = Molecule)) + geom_bar(position= "stack", stat="identity") + 
  scale_x_discrete(guide = guide_axis(angle = -90)) + theme_classic() + scale_fill_manual(values = cols) 

My code for the second figure is 
>cols <- c("DNA" = "darkblue", "RNA" = "cornflowerblue", "Unknown" = "darkgray")
>ggplot(data = nighttable, aes(x = Primary_taxa, y = n, fill = Molecule)) + geom_bar(position= "stack", stat="identity") + 
  scale_x_discrete(guide = guide_axis(angle = -90)) + theme_classic() + scale_fill_manual(values = cols) + scale_y_log10()

<img width="304" alt="Screen Shot 2021-10-08 at 4 05 20 PM" src="https://user-images.githubusercontent.com/73162507/136633193-e8c67d1c-e3e5-49f6-87cd-dfdf97872ce3.png">
<img width="691" alt="Screen Shot 2021-10-08 at 4 05 42 PM" src="https://user-images.githubusercontent.com/73162507/136633217-a4ba2bfb-aab7-45f1-90c0-59dc82fe9a60.png">
<img width="680" alt="Screen Shot 2021-10-08 at 4 05 53 PM" src="https://user-images.githubusercontent.com/73162507/136633227-1b872d2b-28aa-4b43-8033-47435c692b55.png">

**Comment by laurahspencer** [comment, marked as answer]
Have you tried to transform n in your figure, like this-  y = log(n)

**Comment by drlawson** [comment]
<img width="728" alt="Screen Shot 2021-10-10 at 1 03 04 AM" src="https://user-images.githubusercontent.com/73162507/136687674-c8d092d6-06b9-4c8e-92ab-13cac10b2764.png">
Thanks for the suggestion Laura! I actually wanted to transform just the y axis instead of the data. I ended up needing to switch to using plotly instead of ggplot but I finally got it!! :)
